docs/apis-tools/camunda-api-rest/camunda-api-rest-guidelines.md

```diff
@@ -0,0 +1,338 @@
+
+## Data fetching
+
+Most resources provide at least one endpoint to fetch related data. Most of those endpoints provide data with near-real time consistency queried from exported records, if records for the respective resource are exported by the engine. If resources are not based on exported records, for example license data or topology information, the data returned by those endpoints can reflect real time insights or static content.
+
+For most resources, there are search endpoints to query by POST method and a given query request object where applicable. The structure of such query requests always follows the same schema and so does the response, always returning a list of items matching the query criteria.
+
+[Search requests](#search-requests) are forwarded as queries to the datastores that hold the exported records and the query results are returned in the format described in [search responses](#search-responses).
+
+Resources can also support querying subordinate resources. For example, for users and groups, with user search being available at `POST /v2/users/search`, a user's groups can be retrieved using `POST /v2/users/{userKey}/groups/search`. Each resource determines independently if subordinate resources can be accessed this way.
+
+Search endpoints can also be used to directly access entity instances with a unique identifier. As an alternative, resources can also provide GET method endpoints for fetching the data of single instances. In most cases, this is done by a specific key parameter in the URL path, for example `GET /v2/users/{userKey}`.
+
+### Search requests
+
+Query requests consist of the components for **filter**, **sort**, and **page**.
+
+<Tabs groupId="requestComponents" defaultValue="filter" queryString values={[{label: 'Filter', value: 'filter', },{label: 'Sort', value: 'sort', },{label: 'Page', value: 'page', },]} >
+<TabItem value="filter">
+The filter object defines which fields should match. Only items that match the given fields will be returned. The available fields vary by object and are described in the respective search endpoint. Filtering by a unique identifier is usually available in filtering options. Beyond that, the filter options don’t have to comprise all the returned items’ attributes.
+</TabItem>
+<TabItem value="sort">
+The sort object specifies which fields of the object should be sorted and whether they are sorted in ascending (ASC) or descending (DESC) order.
+</TabItem>
+<TabItem value="page">
+The page object details how to slice the result set. An initial search request can omit the page object or define the limit. This specifies the maximum number of results to retrieve per request. Subsequent requests can use the value of the returned firstItemSortValues and lastItemSortValues of the [search responses](#search-responses) to page through the items by copying that array into one of the attributes searchAfter or searchBefore.
```

Comment:
Did we ever figure out a workaround for linking within tabs?

Reply:
@christinaausley do you mean linking from one tab to another? You might be able to use a [query string](https://docusaurus.io/docs/markdown-features/tabs#query-string), assuming the tabs are in the same group (since it's not an anchor, the browser won't scroll).

Reply:
As in, can we link to another page from within a tab. Nevermind though, as I realized this just looked wonky on my local build and looks fine on deploy 👍